duckduckgo/Android
# 383 - Add basic auth support

Task/Issue URL: #383 
Tech Design URL: 
CC: 

**Description**:
Implement the `onReceivedHttpAuthRequest` callback in the `BrowserWebViewClient`.
Add a new dialog that is shown when the client gets a HTTP auth challenge, with supporting string resources.
Add new command to handle authentication requests.
Add tests to cover new functionality.

I've followed the Command pattern that is in use to communicate between the `BrowserTabFragment` and `BrowserTabFragment` and used the `SaveBookmarkDialogFragment` as a template for the new `HttpAuthenticationDialogFragment`.

**Steps to test this PR**:
Successful login:
1. Navigate to http://httpbin.org/basic-auth/test/12345
1. Verify Sign in dialog shown
1. Enter `test` as username, `12345` as password
1. Verify page is shown, content is JSON.

Failed login:
1. Navigate to http://httpbin.org/basic-auth/test/4321
1. Verify Sign in dialog shown
1. Enter `random` as username, `12345` as password
1. Verify auth dialog is shown again

Cancel login:
1. Navigate to http://httpbin.org/basic-auth/test/4321
1. Verify Sign in dialog shown
1. Cancel the dialog
1. Verify page is blank and URL is still http://httpbin.org/basic-auth/test/4321

Success:
![http-auth-success](https://user-images.githubusercontent.com/4399627/54849252-97cf3a00-4cdb-11e9-86fd-d86a3e7cb845.gif)

Failure and cancel:
![http-auth-fail](https://user-images.githubusercontent.com/4399627/54849281-aa497380-4cdb-11e9-955b-bf0dc8e54c5d.gif)



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
